CodeReview.java

```diff
@@ -38,131 +38,151 @@ public int getLargestNumber() {
 
     public int expensiveFunctionExample() {
         int currNum = 0;
-        if (getLargestNumber() > currNum)
-            currNum = getLargestNumber();
-        for (int i=0; i < getLargestNumber(); ++i)
-            System.out.println(getLargestNumber());
+        int largestNum = getLargestNumber();
+
+        if (largestNum > currNum)
+            currNum = largestNum;
+
+        for (int i = 0; i < largestNum; ++i)
+            System.out.println(largestNum);
 
         return currNum;
     }
 
     public void stringExample() {
-        String oneMillionHello = "";
+        StringBuilder oneMillionHelloSB = new StringBuilder();
         for (int i = 0; i < 1000000; i++) {
-            oneMillionHello = oneMillionHello + "Hello!";
+            oneMillionHelloSB.append("Hello!");
```

Comment:
Extract "Hello!" as constant. Otherwise a new string is created each time in the loop iteration